packages/@aws-cdk/aws-appsync/test/appsync-schema.test.ts

```diff
@@ -272,4 +283,108 @@ describe('testing schema definition mode `file`', () => {
+      path: join(__dirname, 'appsync.test.graphql'),
+    });
+    // WHEN
+    new appsync.GraphqlApi(stack, 'API', {
+      name: 'demo',
+      schema: appsync.Schema.fromS3Location(schemaAsset.s3ObjectUrl),
+    });
+
+    // THEN
+    expect(stack).toHaveResourceLike('AWS::AppSync::GraphQLSchema', {
+      DefinitionS3Location: {
+        'Fn::Join': ['', ['s3://',
+          { Ref: 'AssetParametersd915eeba150016e6e51f327f3d1f8a3701dfb7456e8dc17426c8960cecb2d9beS3BucketCB5CC368' },
+          '/',
+          {
+            'Fn::Select': [0, {
+              'Fn::Split': ['||',
+                { Ref: 'AssetParametersd915eeba150016e6e51f327f3d1f8a3701dfb7456e8dc17426c8960cecb2d9beS3VersionKeyB59D7BBD' }],
+            }],
+          },
+          {
+            'Fn::Select': [1, {
+              'Fn::Split': ['||'],
+            }],
+          }]],
+      },
+    });
+  });
+
+  test('definition mode `s3` errors when addType for object is called', () => {
+    // WHEN
+    const api = new appsync.GraphqlApi(stack, 'API', {
+      name: 'demo',
+      schema: appsync.Schema.fromS3Location('garbage'),
+    });
+
+    // THEN
+    expect(() => {
+      api.addType(new appsync.ObjectType('blah', {
+        definition: { fail: t.id },
+      }));
+    }).toThrowError('API cannot add type because schema definition mode is not configured as CODE.');
+  });
+
+  test('definition mode `s3` errors when addType for interface is called', () => {
+    // WHEN
+    const api = new appsync.GraphqlApi(stack, 'API', {
+      name: 'demo',
+      schema: appsync.Schema.fromS3Location('garbage'),
+    });
+
+    // THEN
+    expect(() => {
+      api.addType(new appsync.InterfaceType('blah', {
+        definition: { fail: t.id },
+      }));
+    }).toThrowError('API cannot add type because schema definition mode is not configured as CODE.');
+  });
+
+  test('definition mode `s3` errors when addToSchema is called', () => {
+    // WHEN
+    const api = new appsync.GraphqlApi(stack, 'API', {
+      name: 'demo',
+      schema: appsync.Schema.fromS3Location('garbage'),
+    });
+
+    // THEN
+    expect(() => {
+      api.addToSchema('blah');
+    }).toThrowError('API cannot append to schema because schema definition mode is not configured as CODE.');
+  });
+
+  test('definition mode `s3` errors when addQuery is called', () => {
+    // WHEN
+    const api = new appsync.GraphqlApi(stack, 'API', {
+      name: 'demo',
+      schema: appsync.Schema.fromS3Location('garbage'),
+    });
+
+    // THEN
+    expect(() => {
+      api.addQuery('blah', new appsync.ResolvableField({ returnType: t.string }));
+    }).toThrowError('Unable to add query. Schema definition mode must be CODE. Received: S3');
+  });
+
+  test('definition mode `s3` errors when addMutation is called', () => {
+    // WHEN
+    const api = new appsync.GraphqlApi(stack, 'API', {
+      name: 'demo',
+      schema: appsync.Schema.fromS3Location('garbage'),
+    });
```

Comment:
These lines are repeated in like 5 tests in a row. Can you introduce a `describe()` block with a `beforeEach()` that gets rid of this duplication?